client/src/ui/components/hyperstructures/HyperstructurePanel.tsx

```diff
@@ -47,15 +55,23 @@ export const HyperstructurePanel = ({ entity }: any) => {
   const contributeToConstruction = async () => {
     const formattedContributions = Object.entries(newContributions).map(([resourceId, amount]) => ({
       resource: Number(resourceId),
-      amount,
+      amount: amount * EternumGlobalConfig.resources.resourcePrecision,
     }));
 
-    await contribute_to_construction({
-      signer: account,
-      contributions: formattedContributions,
-      contributor_entity_id: realmEntityId,
-      hyperstructure_entity_id: entity.entity_id,
-    });
+    setIsLoading(Loading.Contribute);
+    setResetContributions(true);
+    try {
+      await contribute_to_construction({
+        signer: account,
+        contributions: formattedContributions,
+        contributor_entity_id: realmEntityId,
+        hyperstructure_entity_id: entity.entity_id,
+      });
+    } finally {
+      setIsLoading(Loading.None);
+      setNewContributions({});
+      setResetContributions(false);
```

Comment:
Wouldn't setting the value to !resetContribution allow us to re-render once instead of twice ?